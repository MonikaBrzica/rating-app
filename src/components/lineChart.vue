<template>
<div class="container" id="chart">
  <div class="title-container">
    <p class="title">Ratings</p>
    <img src="../assets/images/dots-vertical.png"
         alt="...">
  </div>
  <apexchart ref="linechart" type="area" :options="chartOptions" :series="series"></apexchart>

</div>
</template>

<script>
import VueApexCharts from 'vue-apexcharts'
export default {
  name: 'lineChart',
  components: {
    apexchart: VueApexCharts
  },
  data () {
    return {
    }
  },
  computed: {
    statistic () {
      return this.$store.getters.getSumRatingsLine
    },
    series () {
      return [{
        name: 'Very happy',
        data: this.statistic[0]
      }, {
        name: 'happy',
        data: this.statistic[1]
      }, {
        name: 'Meh',
        data: this.statistic[2]
      }, {
        name: 'Sad',
        data: this.statistic[3]
      }, {
        name: 'Very sad',
        data: this.statistic[4]
      }]
    },
    chartOptions () {
      return {
        chart: {
          type: 'area',
          height: '364px',
          foreColor: 'rgba(255,255,255,0.5)',
          zoom: {
            enabled: false
          },
          toolbar: {
            show: false
          },
          animations: {
            enabled: true,
            easing: 'easeinout',
            speed: 900,
            animateGradually: {
              enabled: true,
              delay: 350
            },
            dynamicAnimation: {
              enabled: true,
              speed: 1500
            }
          }
        },
        noData: {
          text: 'No data in database',
          align: 'center',
          verticalAlign: 'middle',
          offsetX: 0,
          offsetY: 0,
          style: {
            color: 'rgba(255,255,255,0.85)',
            fontSize: '20px',
            fontFamily: "'Roboto-Medium', sans-serif"
          }
        },
        grid: {
          position: 'front',
          yaxis: {
            line: {
              show: true
            }
          },
          padding: {
            top: 40,
            right: 24,
            bottom: 28,
            left: 32
          }
        },
        dataLabels: {
          enabled: false
        },
        colors: ['rgb(4, 210, 124)', 'rgb(65, 179, 233)', 'rgb(176, 176, 178)', 'rgb(255, 186, 19)', 'rgb(249, 88, 90)'],
        stroke: {
          curve: 'smooth',
          width: '1'
        },
        xaxis: {
          type: 'datetime',
          categories: this.statistic[5]
        },
        yaxis: {
          align: 'right'
        },
        tooltip: {
          x: {
            format: 'dd/MM/yy HH:mm'
          }
        },
        legend: {
          markers: {
            radius: 0
          }
        },
        markers: {
          colors: ['rgb(4, 210, 124)', 'rgb(65, 179, 233)', 'rgb(176, 176, 178)', 'rgb(255, 186, 19)', 'rgb(249, 88, 90)']
        }
      }
    }
  }
}
</script>

<style scoped lang="scss">
.container{
  width: 100%;
  height:100%;
  .title-container {
    padding: 20px 20px 0px 20px;
    display: flex;
    justify-content: space-between;
    box-sizing: border-box;
    p {
      color: $white-850;
      font-family: 'Roboto-Medium', sans-serif;
      line-height: 20px;
      margin: 0;
    }
    img{
      height: 24px;
    }
  }
}
</style>
