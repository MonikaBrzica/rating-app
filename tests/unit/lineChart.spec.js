import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import lineChart from '../../src/components/lineChart'

const localVue = createLocalVue()
localVue.use(Vuex)
const store = new Vuex.Store({
  getters: {
    getSumRatingsLine (state) {
      const ratings = state.ratings
      const lineChart = [
        [],
        [],
        [],
        [],
        [],
        []
      ]
      if (!ratings) {
        return lineChart
      } else {
        ratings.forEach((elem) => {
          const remind = elem.date.slice(13)
          elem.date = elem.date.replace(remind, ':00:00Z')
          if (lineChart[5].indexOf(elem.date) === -1) {
            lineChart[5].push(elem.date)
          }
        })
        for (let i = 0; i < lineChart.length - 1; i++) {
          for (let j = 0; j < lineChart[5].length; j++) {
            lineChart[i][j] = 0
          }
        }
        ratings.forEach((elem) => {
          for (let i = 0; i < lineChart[5].length; i++) {
            if (elem.date === lineChart[5][i]) {
              lineChart[elem.emojiId - 1][i]++
            }
          }
        })
        return lineChart
      }
    }
  }
})

describe('lineChart', () => {
  const wrapper = shallowMount(lineChart, {
    store,
    localVue
  })
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
