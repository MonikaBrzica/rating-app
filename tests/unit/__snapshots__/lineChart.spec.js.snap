// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lineChart renders correctly 1`] = `
<div
  class="container"
  id="chart"
>
  <div
    class="title-container"
  >
    <p
      class="title"
    >
      Ratings
    </p>
     
    <img
      alt="..."
      src="../assets/images/dots-vertical.png"
    />
  </div>
   
  <apexchart-stub
    class="area"
    height="auto"
    options="[object Object]"
    series="[object Object],[object Object],[object Object],[object Object],[object Object]"
    type="area"
    width="100%"
  />
</div>
`;
